3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/CurrencyConversionEnrichment.scala

```diff
@@ -0,0 +1,205 @@
+
+// Joda-Money
+import org.joda.money.{Money}
+import org.joda.time.{DateTime, DateTimeZone}
+
+// Scala-Forex
+import com.snowplowanalytics.forex._
+import com.snowplowanalytics.forex.oerclient.{OerClientConfig, DeveloperAccount, AccountType, OerResponseError}
+import com.snowplowanalytics.forex.oerclient.OerResponseError._
+import com.snowplowanalytics.forex.{Forex, ForexConfig}
+import com.snowplowanalytics.forex.ForexLookupWhen._
+
+// This project
+import common.utils.ConversionUtils
+import utils.ScalazJson4sUtils
+
+import com.snowplowanalytics.forex.oerclient._
+/**
+* Companion object. Lets us create an CurrencyConversionEnrichment
+* instance from a JValue.
+*/
+object CurrencyConversionEnrichmentConfig extends ParseableEnrichment {
+
+  val supportedSchemaKey = SchemaKey("com.snowplowanalytics.snowplow", "currency_conversion_config", "jsonschema", "1-0-0")
+  
+  //Creates an CurrencyConversionEnrichment instance from a JValue
+  def parse(config: JValue, schemaKey: SchemaKey): ValidatedNelMessage[CurrencyConversionEnrichment] = {
+    isParseable(config, schemaKey).flatMap( conf => {
+      (for {
+        apiKey        <- ScalazJson4sUtils.extract[String](config, "parameters", "apiKey")
+        baseCurrency  <- ScalazJson4sUtils.extract[String](config, "parameters", "baseCurrency")
+        rateAt        <- ScalazJson4sUtils.extract[String](config, "parameters", "rateAt")
+        enrich        =  CurrencyConversionEnrichment(apiKey, baseCurrency, rateAt)
+      } yield enrich).toValidationNel
+    })
+  }
+}
+
+// Object and a case object with the same name
+case class CurrencyConversionEnrichment(
+  apiKey: String,
+  baseCurrency: String, 
+  rateAt: String) extends Enrichment {
+
+  val version = new DefaultArtifactVersion("0.1.0")
+
+  // To provide Validation for org.joda.money.Money variable
+  def eitherToValidation(input:Either[OerResponseError, Money]): Validation[String, String]= {
+    input match {
+      case Right(l) => (l.getAmount().toPlainString()).success[String]
+      case Left(l) => (s"Open Exchange Rates error, message: ${l.errorMessage}").failure[String]
+    }
+  }
+
+  /**
+  * Convert's currency for a given
+  * set of currency, using
+  * Scala-Forex.
+  *
+  * @param trCurrency The desired
+  *        currency for a given 
+  *        amount
+  * @param trAmounts Contains
+  *        total amount, tax, shipping
+  *        amount's
+  * @param tiCurrency Trasaction Item
+  *        Currency
+   * @param tiPrice Trasaction Item
+  *         Price
+  * @return the converted currency
+  *         in TransacrionAmounts
+  *         format and Ttansaction
+  *         Item Price 
+  */
+
+  def convertCurrencies(trCurrency: Option[String], trAmountsTotal: Option[Double], trAmountsTax: Option[Double], trAmountsShipping: Option[Double], tiCurrency: Option[String], tiPrice: Option[Double], collectorTstamp: Option[DateTime]): ValidationNel[String, (String, String, String, String)] = {
```

Comment:
"Amounts" in the variable names ^^ doesn't add anything. trTotal would be fine etc.